web/book/highlight-prql.js

```diff
@@ -35,16 +66,31 @@ formatting = function (hljs) {
     "union",
     "window",
   ];
-  const BUILTIN_FUNCTIONS = ["case", "in", "as"];
-  const KEYWORDS = ["let", "prql", "into"];
+
+  const KEYWORDS = ["let", "prql", "into", "case", "in", "as", "module"];
+
+  const CHAR_ESCAPE = {
+    scope: "char.escape",
+    match: /\\\\|\\([bfnrt]|u\d{4})/,
+  };
+
   return {
     name: "PRQL",
     case_insensitive: true,
     keywords: {
+      built_in: BUILTIN_FUNCTIONS,
       keyword: [...TRANSFORMS, ...BUILTIN_FUNCTIONS, ...KEYWORDS],
-      literal: "false true null ",
+      literal: "false true null",
+      type: DATATYPES,
     },
     contains: [
+      {
+        // docblock
+        begin: "#!",
```

Comment:
Nice

Reply:
One nice addition to this would be changing a comment in the book to use this format (or adding another example), so that in the web book it'll be highlighted as markdown

Reply:
Yeah, I sneaked that one in there, and it should work for at least for headers, bold, italic, and such, but since the token ends when the line ends, it probably doesn't work with multi-line Markdown stuff such as fenced code blocks, lists, and tables.

Also the the PRQL parser doesn't emit any tokens for `#!` so it doesn't get result in anything on the syntax tree which could be used for documentation generation.

I suppose we could extend the Lezer grammar with the [`docComment`](https://lezer.codemirror.net/docs/ref/#highlight.tags.docComment) token, and Ace with the [`block.documentation`](https://github.com/ajaxorg/ace/wiki/Creating-or-Extending-an-Edit-Mode#common-tokens) token.

Reply:
> but since the token ends when the line ends, it probably doesn't work with multi-line Markdown stuff such as fenced code blocks, lists, and tables.

Ah good point...

I guess we could make it though! Basically allow repeated `\s*\n!#`, if that works...